cpu/sam0_common/periph/eth.c

```diff
@@ -185,6 +185,7 @@ static void _init_desc_buf(void)
     /* Initialize TX buffer descriptors */
     for (i=0; i < ETH_TX_BUFFER_COUNT; i++) {
         tx_desc[i].address = (uint32_t) tx_buf[i];
+        tx_desc[i].status = DESC_TX_STATUS_USED;
```

Comment:
Why setting TX buffers to one here ? This is suppose to be done by the IP. Software should only clears this bit.

Reply:
The bit gets set by the hardware when it's done with the buffer.
So this will mark all buffers as unused to the driver.

Reply:
But GMAC IP will not find an available buffer when it want to send data if they are all marked as used already.
GMAC IP is supposed to take an unused buffer (mark == 0) then take it (mark it to one to tell software this buffer is being used)
This is not software job to set it.

Reply:
The bit is [set to 0](https://github.com/RIOT-OS/RIOT/blob/c067fd2801a5f2e3ec3c9983736b4069cbfd770c/cpu/sam0_common/periph/eth.c#L253) before starting the transmission.

Reply:
Alright, I missed the inversion in the if statement for the `thread_yield()`
That's a bit weird but it makes sense.

So do we  want really want to block this or drop the new frames (like this is the case for RX) ?